### ISSUE TITLE
ENOAuthViewController does not catch OAuth token in webView:shouldStartLoadWithRequest:navigationType:.

### ISSUE THREAD

**Issue by sonsongithub** [Evernote/evernote-sdk-ios#32]
In OAuth authorization page, UIWebView does not a catch URL which begin with en-(id?)://response?oauth_token=(token)&oauth_verifier=(verifier) after user's authentication.
I peeked request URL in webView:shouldStartLoadWithRequest:navigationType:., but I counld n't find such as URL.

It's my account's or codes' bug? or server's specification has been changed.....?

My workaround is that to make UIWebViewDelegate catch othe URL such as https://sandbox.evernote.com/Home.action?en-(id?)://response?oauth_token=(token)&oauth_verifier=(verifier) by the following code.

``` Objective-C
- (BOOL)webView:(UIWebView *)webView shouldStartLoadWithRequest:(NSURLRequest *)request navigationType:(UIWebViewNavigationType)navigationType
{
    NSLog(@"%@", [request.URL absoluteString]);
    if ([[request.URL absoluteString] hasPrefix:self.oauthCallbackPrefix]) {
        // this is our OAuth callback prefix, so let the delegate handle it
        if (self.delegate) {
            [self.delegate oauthViewController:self receivedOAuthCallbackURL:request.URL];
        }
        return NO;
    }
    else if ([[request.URL query] hasPrefix:self.oauthCallbackPrefix]) {
        // this is our OAuth callback prefix, so let the delegate handle it
        if (self.delegate) {
            NSURL *newURL = [NSURL URLWithString:request.URL.query];
            [self.delegate oauthViewController:self receivedOAuthCallbackURL:newURL];
        }
        return NO;
    }
    return YES;
}
```

**Comment by sonsongithub** [comment]
This problem happens when consumer key includes "_"(underscore)...?

**Comment by mustafa** [comment]
The URL that the code is supposed to be intercept is of the form : "en-<consumer id>://response?oauth_token=<oauth token>&oauth_verifier=<oauth verifier>" . This seems to work correctly.

**Comment by sonsongithub** [comment]
No, it does not.
I guess that Evernote does not return "en-(id?)://response?oauth_token=(token)&oauth_verifier=(verifier) after user's authentication." when the consumer key includes "_"(under score) empirically.

**Comment by uma-blue** [comment]
I'm looking forward to official-launch this.

**Comment by mustafa** [comment]
Thanks, we will fix this in an upcoming release on the server. For now, I recommend creating a new api key without an underscore.

**Comment by mustafa** [comment]
Fixed in 1.0.0

**Comment by uma-blue** [comment]
Thanks, Fixed!
